internal/fetch/fetch.go

```diff
@@ -27,7 +28,7 @@ func Fetch(ctx context.Context, hash, target, folder string, insecure bool) erro
 	case "oci":
 		return oci.Fetch(ctx, hash, newTarget, folder, insecure)
 	case "s3":
-		return fmt.Errorf("s3 schema not (yet) supported: %s", target)
+		return s3.DownloadFromS3(ctx, hash, newTarget, folder)
```

Comment:
For the other implementation I used `Fetch` and `Upload` for the function name (to be consistent). Could we do the same here ?